addressfield.webform.inc

```diff
@@ -0,0 +1,484 @@
+ * @param mixed $value
+ *   If editing an existing submission or resuming a draft, this will contain
+ *   an array of values to be shown instead of the default in the component
+ *   configuration. This value will always be an array, keyed numerically for
+ *   each value saved in this field.
+ * @param bool $filter
+ *   Whether or not to filter the contents of descriptions and values when
+ *   rendering the component. Values need to be unfiltered to be editable by
+ *   Form Builder.
+ *
+ * @return array
+ *   Form element.
+ *
+ * @see _webform_client_form_add_component()
+ */
+function _webform_render_addressfield($component, $value = NULL, $filter = TRUE) {
+  $element = array(
+    '#type'             => 'fieldset',
+    '#title'            => $filter ? webform_filter_xss($component['name']) : $component['name'],
+    '#title_display'    => $component['extra']['title_display'] ? $component['extra']['title_display'] : 'before',
+    '#attributes'       => $component['extra']['attributes'],
+    '#theme_wrappers'   => array('webform_element'),
+    '#description'      => $filter ? webform_filter_descriptions($component['extra']['description']) : $component['extra']['description'],
+    '#required'         => $component['required'],
+    '#weight'           => $component['weight'],
+    '#translatable'     => array(
+      'title',
+      'description',
+    ),
+  );
+
+  $available = !empty($component['extra']['available_countries']) ? $component['extra']['available_countries'] : NULL;
+
+  // Get the current address.
+  $address = _addressfield_expand_value($value);
+  if (empty($address)) {
+    if (!empty($component['value'])) {
+      $address = $component['value'];
+    }
+    else {
+      $address = _webform_addressfield($component['cid']);
+    }
+  }
+  if (empty($address)) {
+    $address = _webform_addressfield_default_values($available, $component);
+  }
+
+  // Generate the address form.
+  $context = array(
+    'mode' => 'form',
+    'instance' => array(
+      'required' => $component['required'],
+    ),
+    'form_key' => $component['form_key'],
+  );
+  $handlers = !empty($component['extra']['format_handlers']) ? $component['extra']['format_handlers'] : array('address');
+  $element += addressfield_generate($address, $handlers, $context);
+
+  if (empty($address['country'])) {
+    $element['street_block']['#access'] = FALSE;
+    $element['locality_block']['#access'] = FALSE;
+  }
+
+  if (isset($element['country'])) {
+    if (!empty($available)) {
+      $element['country']['#options'] = array_intersect_key($element['country']['#options'], $available);
+
+      // Hide the country element only if there is one option and the whole field
+      // is required, otherwise there will always be an additional None option.
+      // @see addressfield_format_address_hide_country()
+      if (!empty($handlers['address-hide-country']) && count($element['country']['#options']) == 1 && $component['required']) {
```

Comment:
```suggestion
      if (!empty($handlers['address_hide_country']) && count($element['country']['#options']) == 1 && $component['required']) {
```

Reply:
Thank you, @yorkshire-pudding! Fixed and tested - works well.